plugin/hook/adapter_test.go

```diff
@@ -0,0 +1,622 @@
+					},
+				},
+			},
+		})
+		expected := models.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue: "",
+					Questions: []models.PluginQuestion{
+						{
+							Name:         "nil",
+							Prompt:       "nil",
+							Help:         "nil",
+							Default:      "nil",
+							Multiselect:  nil,
+							SubQuestions: []models.PluginSubQuestion{},
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non empty inputs", func(t *testing.T) {
+		actual := adaptQuestionFromProto(&pb.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []*pb.PluginQuestion_SubQuestion{
+				{
+					IfValue: "",
+					Questions: []*pb.PluginQuestion{
+						{
+							Name:         "key",
+							Prompt:       "sec",
+							Help:         "this",
+							Default:      "do",
+							Multiselect:  []string{"a", "ba", "b"},
+							SubQuestions: []*pb.PluginQuestion_SubQuestion{},
+						},
+					},
+				},
+			},
+		})
+		expected := models.PluginQuestion{
+			Name:        "key",
+			Prompt:      "sec",
+			Help:        "this",
+			Default:     "do",
+			Multiselect: []string{"a", "ba", "b"},
+			SubQuestions: []models.PluginSubQuestion{
+				{
+					IfValue: "",
+					Questions: []models.PluginQuestion{
+						{
+							Name:         "key",
+							Prompt:       "sec",
+							Help:         "this",
+							Default:      "do",
+							Multiselect:  []string{"a", "ba", "b"},
+							SubQuestions: []models.PluginSubQuestion{},
+						},
+					},
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+}
+
+func Test_adaptConfigsToProto(t *testing.T) {
+	t.Run("should work for empty input", func(t *testing.T) {
+		actual := adaptConfigsToProto(nil)
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for undefined input", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "",
+					Value: "",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for empty Configs input", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "",
+				Value: "",
+			},
+		})
+		expected := &pb.HookConfigs{
+			Configs: []*pb.HookConfigs_Config{
+				{
+					Name:  "",
+					Value: "",
+				},
+			},
+		}
+		assert.Equal(t, expected, actual)
+	})
+	t.Run("should work for non-empty input", func(t *testing.T) {
+		actual := adaptConfigsToProto(models.HookPluginConfigs{
+			models.HookPluginConfig{
+				Name:  "nil",
```

Comment:
Remove all usages of `"nil"` that string containing `nil` word as it is confusing.

Reply:
done